poetry/vcs/git.py

```diff
@@ -9,7 +9,7 @@
 
 pattern_formats = {
     "protocol": r"\w+",
-    "user": r"[a-zA-Z0-9_.-]+",
+    "user": r"[a-zA-Z0-9_.\+-]+(:[\w\d]+)?",
```

Comment:
I found a case in my usage where this doesn't appear to be enough. The deploy key for gitlab appears to allow `-` characters, and the regex portion `[\w\d]+` doesn't match when a `-` is present.

Changing the `[\w\d]+` to ``[\w\d-]+` got it working for me.

Reply:
Thanks for the feedback. Is there any source where it is described how the deployment key looks like? I wasn't able to find one :(

Reply:
I haven't found any. The only suggestion I can make is to perhaps make the portion after the `:` much less restrictive since it looks like the end of that match will be bounded by `@` in all the patterns below.

I don't know what value there is in trying to restrict what can go after the `:` unless there's a need for it to make the matching work properly.